projects/storefrontlib/layout/theme/theme.service.ts

```diff
@@ -27,10 +27,10 @@ export class ThemeService {
     // Theme value is a string. It is put in the generic multi-value
     // property of the SiteContextConfig. So the array's first item
     // is the theme value.
-    this.setTheme(this.config.context[THEME_CONTEXT_ID]?.[0]);
+    this.setTheme(this.config.context?.[THEME_CONTEXT_ID]?.[0]);
   }
 
-  setTheme(theme: string): void {
+  setTheme(theme?: string): void {
```

Comment:
I don't think we should make theme optional. Similar reason as [this](https://github.com/SAP/spartacus/pull/15658/files#r870749898)